### PR TITLE
Modifications

### PR DESCRIPTION
**Pull request by TuomasBorman**
I splitted this PR since it had too many things

https://github.com/microbiome/mia/pull/298

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -34,6 +34,10 @@
   is.character(x) && length(x) == 1L
 }
 
+.is_an_integer <- function(x){
+    is.numeric(x) && length(x) == 1L && x%%1==0
+}
+
 .are_whole_numbers <- function(x){
   tol <- 100 * .Machine$double.eps
   abs(x - round(x)) <= tol && !is.infinite(x)
```
Comment (TuomasBorman) [inline R/utils.R:43]: .is_an:integer expects a single value, .are_whole_numbers not

File: R/getExperimentCrossAssociation.R [hunk 2]
```diff
@@ -607,11 +607,9 @@ setMethod("getExperimentCrossCorrelation", signature = c(x = "ANY"),
...
-    if( !(class(obj) == "TreeSummarizedExperiment" || 
-          class(obj) == "SummarizedExperiment") ){
+    if( !(is(obj, "SummarizedExperiment")) ){
         stop("The class of experiment specified by ", 
-             deparse(substitute(experiment)), " must be 'TreeSummarizedExperiment' ",
-             "or 'SummarizedExperiment'.",
+             deparse(substitute(experiment)), " must be 'SummarizedExperiment'.",
              call. = FALSE)
     }
 }
```
Comment (TuomasBorman) [inline R/getExperimentCrossAssociation.R:615]: Made the test simpler

File: R/getExperimentCrossAssociation.R [hunk 3]
```diff
@@ -622,39 +620,19 @@ setMethod("getExperimentCrossCorrelation", signature = c(x = "ANY"),
...
         tse <- altExp(tse, altExp)
-        return(tse)
-    # Otherwise give error
-    } else{
-        stop("'", deparse(substitute(altExp)), "' must be NULL, or a single numeric or ",
-             "character value specifying altExp of experiment ", exp_num, ".",
-             call. = FALSE)
     }
+    return(tse)
 }
```
Comment (TuomasBorman) [inline R/getExperimentCrossAssociation.R:636]: Tets moved to utils since it is used in more than one function

File: R/runDPCoA.R [hunk 4]
```diff
@@ -97,19 +97,34 @@ setGeneric("calculateDPCoA", signature = c("x", "y"),
...
+    # Check ncomponents
+    if( !(.is_an_integer(ncomponents) && ncomponents > 0) ){
+        stop("'ncomponents' must be a single integer value specifying the number ",
+             "of DPCoA dimensions.", call. = FALSE)
+    }
+    # Check ntop
+    if( !(is.null(ntop) || (.is_an_integer(ntop) && ntop > 0))  ){
+        stop("'ntop' must be NULL or a single integer value specifying the number ",
+             "of features with the highest variance.", call. = FALSE)
+    }
```
Comment (TuomasBorman) [inline R/runDPCoA.R:109]: There we no tests

File: R/runDPCoA.R [hunk 5]
```diff
@@ -168,11 +183,21 @@ setMethod("calculateDPCoA", signature = c("TreeSummarizedExperiment","missing"),
 #' @rdname runDPCoA
 #' @importFrom SingleCellExperiment reducedDim<-
 runDPCoA <- function(x, ..., altexp = NULL, name = "DPCoA"){
-    if (!is.null(altexp)) {
+    # Input check
+    # Check and get altExp if it is not NULL
+    if( !is.null(altexp) ){
+        # Check altExp
+        .check_altExp_present(altexp, x)
+        # Get altExp
```
Comment (TuomasBorman) [inline R/runDPCoA.R:191]: Check for altexp

File: R/runDPCoA.R [hunk 6]
```diff
@@ -168,11 +183,21 @@ setMethod("calculateDPCoA", signature = c("TreeSummarizedExperiment","missing"),
 #' @rdname runDPCoA
 #' @importFrom SingleCellExperiment reducedDim<-
 runDPCoA <- function(x, ..., altexp = NULL, name = "DPCoA"){
```
Comment (TuomasBorman) [inline R/runDPCoA.R:185]: Here alternative experiment is `altexp`. We also use `altExp`, e.g.,, in `getExperimentCrossCorrelation` . We should choose only one. --> I think altExp is better

File: R/runDPCoA.R [hunk 7]
```diff
@@ -168,11 +183,21 @@ setMethod("calculateDPCoA", signature = c("TreeSummarizedExperiment","missing"),
...
+        # Get altExp
         y <- altExp(x, altexp)
     } else {
         y <- x
     }
+    # Check name
+    if( !.is_a_string(name) ){
+        stop("'name' must be a single character value specifying a name of ",
+             "reducedDim where the result will be stored.", call. = FALSE)
+    }
```
Comment (TuomasBorman) [inline R/runDPCoA.R:200]: There were no test for the name

File: tests/testthat/test-5runDPCoA.R [hunk 8]
```diff
@@ -33,32 +33,31 @@ test_that("runDPCoA", {
...
+    )
+    expect_error(
+        runDPCoA(esophagus, name = c("test", "test2"), assay_name = "counts",
+                 tree_name = "phylo", ncomponents = 1.3, ntop = "test",
+                 subset_row = NULL, scale = FALSE, transposed = FALSE)
+    )
+    expect_error(
+        runDPCoA(esophagus, name = 1, assay_name = "counts", tree_name = "phylo", ncomponents = 1.3, ntop = "test",
+                 subset_row = NULL, scale = FALSE, transposed = FALSE)
+    )
```
Comment (TuomasBorman) [inline tests/testthat/test-5runDPCoA.R:60]: Enable these tests

**Comment by TuomasBorman** [comment]
I think this is ok to be merged after altExp thing is decided (whether to use altexp or altExp)

**Comment by antagomir** [comment]
I do not really have a definite opinion on this. It might be confusing to use two different spelling forms but if they are in different context (data object vs function/argument name) then I think it can be acceptable. Do we have examples from single cell tools how to have dealt with this?

**Comment by TuomasBorman** [comment]
They actually use` altexp` in scater...

https://bioconductor.org/packages/release/bioc/manuals/scater/man/scater.pdf

Should we unify the naming scheme

**Comment by antagomir** [comment]
Would be good to use the same unless we have strong reasons to deviate.

**Comment by TuomasBorman** [comment]
I merge this and let's solve altExp thing in https://github.com/microbiome/mia/pull/311

**Pull request merged**